pkg/common/configs/configvalidator.go

```diff
@@ -595,6 +596,26 @@ func checkLimits(limits []Limit, obj string, queue *QueueConfig) error {
 	return nil
 }
 
+func checkLimitsStructure(partitionConfig *PartitionConfig) error {
+	partitionLimits := partitionConfig.Limits
+	rootQueue := partitionConfig.Queues[0]
+
+	if len(partitionConfig.Queues) < 1 || strings.ToLower(partitionConfig.Queues[0].Name) != RootQueue {
+		return fmt.Errorf("top queue is not root")
+	}
+
+	if len(partitionLimits) > 0 && len(rootQueue.Limits) > 0 && !reflect.DeepEqual(partitionLimits, rootQueue.Limits) {
+		return fmt.Errorf("partition limits and root queue limits are not equivalent")
+	}
+
+	// if root queue limits not defined, apply partition limits
+	if len(partitionLimits) > 0 && len(rootQueue.Limits) == 0 {
+		partitionConfig.Queues[0].Limits = partitionLimits
```

Comment:
ditto

Reply:
Actually this can't be replaced with `rootQueue`, otherwise the partition limits won't change.

Reply:
oh, I get it now.

Reply:
ping @chia7712 , already addressed comments in the latest comment. Many thanks~